File: fictionManagement/src/main/java/com/ttn/fictionManagement/entity/Fiction.java
package com.ttn.fictionManagement.entity;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "fiction")
public class Fiction {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "countView")
    private Integer countView;

    @Column(name = "coverURL")
    private String coverUrl;

    @Column(name = "status")
    private boolean status;

    @Column(name = "description")
    private String description;

    @Column(name = "userId")
    private Integer userId;
}

File: fiction-management-fe/src/app/interface/fiction.ts
import { Chapter } from './chapter';
import { Comment } from './comment';
import { Process } from './process';
import { Tag } from './tag';
import { User } from './user';

export interface Fiction {
  id?: number;
  name: string;
  countView: number;
  coverUrl: string;
  status: boolean;
  description: string;
  userId: number;
  user?: User;
  tags?: Tag[];
  process?: Process;
  chapters?: Chapter[];
  comments?: Comment[];
  coverPublicId?: string;
}
